src/Microsoft.Health.Fhir.Anonymizer.Shared.Core/Processors/GeneralizeProcessor.cs

```diff
@@ -0,0 +1,158 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using EnsureThat;
+using Hl7.Fhir.ElementModel;
+using Hl7.Fhir.Model;
+using Hl7.FhirPath;
+using Hl7.Fhir.Model.Primitives;
+using Microsoft.Health.Fhir.Anonymizer.Core.Models;
+using Microsoft.Health.Fhir.Anonymizer.Core.Processors.Settings;
+using Microsoft.Health.Fhir.Anonymizer.Core.AnonymizerConfigurations;
+
+namespace Microsoft.Health.Fhir.Anonymizer.Core.Processors
+{
+    public partial class GeneralizeProcessor : IAnonymizerProcessor
+    {
+        private static readonly HashSet<string> s_integerValueTypeNames = new HashSet<string>
+        {
+            FHIRAllTypes.Integer.ToString(),
+            FHIRAllTypes.PositiveInt.ToString(),
+            FHIRAllTypes.UnsignedInt.ToString()
+        };
+
+        private static readonly HashSet<string> s_DateOrTimeValueTypeNames = new HashSet<string>
+        {
+            FHIRAllTypes.DateTime.ToString(),
+            FHIRAllTypes.Date.ToString(),
+            FHIRAllTypes.Time.ToString(),
+            FHIRAllTypes.Instant.ToString()
+        };
+
+        private void GeneralizeIntegerNode(ElementNode node, object targetValue)
+        {
+            //Transform the target value into int32.
+            int target;
+            try
+            {
+                target = Convert.ToInt32(targetValue);
+            }
+            catch (Exception ex)
+            {
+                throw new AnonymizerConfigurationErrorsException($"Invalid data types or format of expression {targetValue} output", ex);
+            }
+
+            //Check target value for positive integer and unsigned interger.
+            if (string.Equals(FHIRAllTypes.PositiveInt.ToString(), node.InstanceType, StringComparison.InvariantCultureIgnoreCase) && target <= 0)
```

Comment:
Make them into IsPositiveIntNode and IsUnsignedIntNode to keep consistent with IsTimeNode